packages/itwin/breakdown-trees/src/Views/visibility/DataLink.ts

```diff
@@ -10,100 +10,79 @@ export class DataLink {
   public static async queryAllBuildings(iModel: IModelConnection) {
     const query = `SELECT building.ECInstanceId as id, building.UserLabel as label, building.CodeValue as code
     FROM BuildingSpatial:Building building`;
-    const rows = await this.executeQuery(iModel, query);
 
-    return rows;
+    return this.executeQuery(iModel, query);
   }
 
   // This doesn't work on Revit derived data because:
   //  a. Too many stories (need to limit to those with rooms) and
   //  b. No composingElement (because no buildings)
   public static async queryAllStories(iModel: IModelConnection) {
     const query = `SELECT story.ECInstanceId as id, story.UserLabel as label, story.CodeValue as code, building.ECInstanceId as composingId FROM BuildingSpatial:Story story JOIN BuildingSpatial.Building building ON building.ECInstanceId = story.composingElement.id`;
-    const rows = await this.executeQuery(iModel, query);
 
-    return rows;
+    return this.executeQuery(iModel, query);
   }
 
   // Use this for Revit derived data to work around issues with queryAllStories
   public static async queryStoriesWithRooms(iModel: IModelConnection) {
     const query = `SELECT DISTINCT story.ECInstanceId as id, story.UserLabel as label, story.CodeValue as code
     FROM BuildingSpatial:Story story, BuildingSpatial:Space room WHERE room.composingElement.id = story.ECInstanceId`;
-    const rows = await this.executeQuery(iModel, query);
-    return rows;
+
+    return this.executeQuery(iModel, query);
   }
 
   public static async queryStoryRange(iModel: IModelConnection, storyId: string, clipAtSpaces: boolean): Promise<Range3d | undefined> {
```

Comment:
there are significant changes in this function, but we seem to be only testing that "it calls query() and it returns whatever it gets"

could you add a test for "storyid" and "clipatspaces" parameters too?

Reply:
Added tests

Reply:
can we also have a test to capture the other point? "if that query returns undefined then we need to run the query again with CompositeOverlapsSpatialElements"

also, i think it would be better not to have a constructquery function in the test, but just hardcode the expected string.

Reply:
Added new test to capture the mentioned scenario + refractor